src/vm/wren_core.wren

```diff
@@ -302,6 +302,52 @@ class List is Sequence {
     }
     return result
   }
+
+  sort() { sort_(this.map {|n| [n, n] }.toList) }
+
+  sort(key) { sort_(this.map {|n| [n, key.call(n)] }.toList) }
```

Comment:
Calling the key function on comparison instead of duplicating and then converting back in to single elements took the time down to 1.459349 secs.  Having a sort in place would also be a good idea.